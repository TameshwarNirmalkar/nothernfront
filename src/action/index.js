export const FETCH_DATA = 'FETCH_DATA';
export const API_ERROR = 'API_ERROR';


const fetchPokemonSuccess = (state) => {
    return {
        type: FETCH_DATA,
        pokemonList: state,
        pending: true
    };
}

const fetchPokemonError = (state) => {
    return {
        type: API_ERROR,
        ...state,
        pending: false
    };
}
// default function to display redux action format
export const getPokemon = () => {
    return dispatch => {
        
        fetch('https://pokeapi.co/api/v2/pokemon')
            .then(res => res.json())
            .then(res => {
                if (res.error) {
                    throw (res.error);
                }
                dispatch(fetchPokemonSuccess(res.results));
                return res.results;
            })
            .catch(error => {
                dispatch(fetchPokemonError(error));
            })
    };
}
