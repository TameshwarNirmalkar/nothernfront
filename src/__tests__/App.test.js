import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import PokemonList from '../components/pokemonList';
import configureStore, { createMockStore } from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import { getPokemon, FETCH_DATA } from '../action';


describe('>>>APP --- REACT-REDUX (Shallow + passing the {store} directly)', () => {
    const pokemonList = {
        account: {
            pokemonList: [
                { name: "bulbasaur", url: "https://pokeapi.co/api/v2/pokemon/1/" },
                { name: "ivysaur", url: "https://pokeapi.co/api/v2/pokemon/2/" },
                { name: "venusaur", url: "https://pokeapi.co/api/v2/pokemon/3/" },
            ]
        },
        fetchData: jest.fn()
    };
    const initialState = { ...pokemonList };
    const mockStore = configureStore();
    let store, container, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);
        // container = mount(<Provider store={store}><PokemonList dataList={pokemonList} /></Provider>);
        // wrapper = shallow(<App store={store} />)
    });

    it('+++ render the PokemonList component', () => {
        // const str = createMockStore([])({});
        const tree = create(
            <Provider store={store}>
                <PokemonList dataList={initialState.account.pokemonList} />
            </Provider>,
        );

        expect(tree).toMatchSnapshot();
    });

    

});
