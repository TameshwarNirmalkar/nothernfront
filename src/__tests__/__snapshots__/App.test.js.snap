// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>APP --- REACT-REDUX (Shallow + passing the {store} directly) +++ render the PokemonList component 1`] = `
<div
  className="container mt-5"
>
  <div
    className="carousel"
  >
    <div
      aria-live="polite"
      className="horizontalSlider___281Ls carousel__slider carousel__slider--horizontal"
      onKeyDown={[Function]}
      role="listbox"
      style={Object {}}
      tabIndex={0}
    >
      <div
        className="carousel__slider-tray-wrapper carousel__slider-tray-wrap--horizontal"
        style={Object {}}
      >
        <ul
          className="sliderTray___-vHFQ sliderAnimation___300FY carousel__slider-tray carousel__slider-tray--horizontal"
          onClickCapture={[Function]}
          onMouseDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "transform": "translateX(0%) translateX(0px)",
              "width": "100%",
            }
          }
        >
          <li
            aria-selected={true}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "41.666666666666664%",
                "width": "33.333333333333336%",
              }
            }
            tabIndex={0}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
            >
              <h2
                className="panel-heading"
              >
                <span>
                  bulbasaur
                </span>
                <span>
                   ID: 
                  1
                </span>
              </h2>
              <div
                className="d-flex justify-content-center"
              >
                <img
                  alt="bulbasaur"
                  className="w-100 h-100"
                  src="no-image.png"
                />
              </div>
            </div>
          </li>
          <li
            aria-selected={true}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "41.666666666666664%",
                "width": "33.333333333333336%",
              }
            }
            tabIndex={0}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
            >
              <h2
                className="panel-heading"
              >
                <span>
                  ivysaur
                </span>
                <span>
                   ID: 
                  2
                </span>
              </h2>
              <div
                className="d-flex justify-content-center"
              >
                <img
                  alt="ivysaur"
                  className="w-100 h-100"
                  src="no-image.png"
                />
              </div>
            </div>
          </li>
          <li
            aria-selected={true}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "41.666666666666664%",
                "width": "33.333333333333336%",
              }
            }
            tabIndex={0}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
            >
              <h2
                className="panel-heading"
              >
                <span>
                  venusaur
                </span>
                <span>
                   ID: 
                  3
                </span>
              </h2>
              <div
                className="d-flex justify-content-center"
              >
                <img
                  alt="venusaur"
                  className="w-100 h-100"
                  src="no-image.png"
                />
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="prev-next"
    >
      <button
        aria-label="previous"
        className="buttonBack___1mlaL carousel__back-button button-cls"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Prev
      </button>
      <button
        aria-label="next"
        className="buttonNext___2mOCa carousel__next-button button-cls"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
